tests/generate_small_sample.py

```diff
@@ -0,0 +1,133 @@
+from test_main import TestFiles
+from pathlib import Path
+import hashlib
+
+from pysource_minimize import minimize
+import sys
+import subprocess as sp
+import textwrap
+import os
+import linecache
+from executing import Source
+from multiprocessing import Pool,get_context
+import tempfile
+import hashlib
+import time
+
+last_samples_dir = Path(__file__).parent / "last_samples"
+last_samples_dir.mkdir(exist_ok=True)
+samples_dir = Path(__file__).parent / "samples"
+
+
+# big_samples = (Path(__file__).parent / "samples").glob("*.py")
+# big_samples = list(Path.home().rglob("*.py"))
+small_samples = Path(__file__).parent / "small_samples"
+
+
+def big_samples():
+    yield from last_samples_dir.rglob("*.py")
+    yield from samples_dir.rglob("*.py")
```

Comment:
Aren't these very likely to be under `Path.home()`?

Reply:
> yes, very likely. But checking all files under home takes for me something like 4 hours and the samples are some ~2 minutes maybe. The idea here is to search a set of files with a likely hit first.

Ah, it's about order. Makes sense.

> But: The thing is that searching home is maybe not the best idea (not deterministic, kind of private data ...). One Idea I had was to create a repository on github with submodules. Each submodule should be a python project with "crazy" code pypy cpython every project which depends on executing ... what do you think?

If you want to do something like that, I think it'd be better to just have a repo with copies of all the Python files from the various projects. No need to mess around with git submodules, no need to keep them up to date, no need for all the non-python files.